hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/cache/FullTableCache.java

```diff
@@ -111,18 +116,31 @@ public void loadInitial(CacheKey<KEY> key, CacheValue<VALUE> value) {
   @Override
   public void put(CacheKey<KEY> cacheKey, CacheValue<VALUE> value) {
     try {
-      lock.writeLock().lock();
+      lock.readLock().lock();
```

Comment:
Currently, there are no parallel calls to `TableCache#put` method. What will happen if someone makes a parallel put calls with same key? Will we end up in inconsistent state here with `readLock`?

Reply:
Currently no use case, but if someone do parallel call, the last one will get updated as collection is concurrentSkipListMap and no corruption.
This is same case for PartialCacheTable where same logic, but no lock. Lock purpose was eviction issue as reported in [HDDS-4583](https://issues.apache.org/jira/browse/HDDS-4583) Jira where target problem is mentioned. And hence this is not required.